### PR TITLE
Implementation of -lp alias for -LiteralPath Parameter #6732

### PR DESCRIPTION
**Pull request by kvprasoon**
Implementation of -lp alias for -LiteralPath variable for issue #6732 <br> <br>
**PR Summary** <br>
Added test case for `-lp` alias usage for below cmdlets <br>

`Add-Content ` <br>
`Add-Type` <br>
`Clear-Content` <br>
`Clear-Item` <br>
`Clear-ItemProperty` <br>
`Convert-Path` <br>
`Copy-Item` <br>
`Export-Alias` <br>
`Export-Clixml` <br>
`Export-Csv` <br>
`Export-FormatData` <br>
`Format-Hex` <br>
`Get-ChildItem` <br>
`Get-Content` <br>
`Get-FileHash` <br>
`Get-Item` <br>
`Get-ItemProperty` <br>
`Get-ItemPropertyValue` <br>
`Import-Alias` <br>
`Import-Clixml` <br>
`Import-Csv` <br>
`Import-PowerShellDataFile` <br>
`Move-Item` <br>
`Out-File` <br>
`Push-Location` <br>
`Remove-Item` <br>
`Rename-Item` <br>
`Resolve-Path` <br>
`Select-String` <br>
`Select-Xml` <br>
`Set-Content` <br>
`Set-Item` <br>
`Set-ItemProperty` <br>
`Set-Location` <br>
`Split-Path` <br>
`Start-Job` <br>
`Start-Transcript` <br>
`Tee-Object` <br>
`Test-Path` <br>
`Unblock-File` <br>

**PR Checklist** <br>
 
- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission) <br>
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission) <br>
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes) <br>
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission) <br>
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress). <br>
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready. <br>
- **User-facing changes** <br>
    - [x] Not Applicable <br>
    - **OR** <br>
    - [ ] User-facing [Documentation needed] <br>(https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission) <br>
        - [ ] Issue filed - Issue link: <br>
- **Testing - New and feature** <br>
    - [x] Not Applicable or can only be tested interactively
    - **OR** <br>
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting) <br>
        - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr) <br>

### CODE REVIEW

**Comment by kvprasoon** [comment]
Fixed unit test failures for
`Get-Content`
`Set-Content`
`Move-Item`
`Rename-Item`
`Tee-Object`
`Add-Type`
cmdlets

File: PowerShell.Common.props [hunk 1]
```diff
@@ -13,8 +13,8 @@
   <Target Name="GetPSCoreVersionFromGit"
     BeforeTargets="_GenerateRestoreProjectSpec;GenerateNuspec;BeforeBuild"
   >
-
-    <Exec Command='git describe --abbrev=60 --long'
+  
+    <Exec Command='git describe --abbrev=60 --long'
```
Comment (iSazonov) [inline PowerShell.Common.props]: Please revert changes in the file.

File: test/powershell/Modules/Microsoft.PowerShell.Core/Job.Tests.ps1 [hunk 2]
```diff
@@ -8,6 +8,13 @@ Describe "Job Cmdlet Tests" -Tag "CI" {
         AfterEach {
             Get-Job | Remove-Job -Force
         }
+
+        It "should not throw exception when -LP alias for LiteralPath is used " {
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Core/Job.Tests.ps1]: Please remove all alias tests - we have already special tests for aliases and no need to add them here.

**Review by iSazonov — COMMENTED** [review]
Leave a comment

File: test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1 [hunk 3]
```diff
@@ -357,4 +357,13 @@ Describe "Get-Content -Raw test" -Tags "CI" {
...
     }
+<<<<<<< HEAD
+
+    It "should not throw exception when -LP alias for LiteralPath is used " {
+        'test' | Out-File -FilePath $TestDrive\TestTile.txt
+        { Get-Content -lp $TestDrive\TestTile.txt } | Should -Not -Throw
+      } 
+    
+=======
+>>>>>>> parent of d0002ef6... Implementation of -lp alias for -LiteralPath variable #6732
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Get-Content.Tests.ps1]: Please fix.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Move-Item.Tests.ps1 [hunk 4]
```diff
@@ -14,6 +14,13 @@ Describe "Move-Item tests" -Tag "CI" {
         "$target" | Should -FileContentMatchExactly "This is content"
     }
 
+<<<<<<< HEAD
+    It "should not throw exception when -LP alias for LiteralPath is used " {
+        { Move-Item -lp  $target -Destination $source-WhatIf } | Should -Not -Throw
+    } 
+
+=======
+>>>>>>> parent of d0002ef6... Implementation of -lp alias for -LiteralPath variable #6732
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Move-Item.Tests.ps1]: Please fix.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Rename-Item.Tests.ps1 [hunk 5]
```diff
@@ -13,4 +13,11 @@ Describe "Rename-Item tests" -Tag "CI" {
         test-path $target | Should -BeTrue
         "$target" | Should -FileContentMatchExactly "This is content"
     }
+<<<<<<< HEAD
+
+    It "should not throw exception when -LP alias for LiteralPath is used " {
+        { Rename-Item -lp $target -NewName 'NewName' -WhatIf } | Should -Not -Throw
+      }     
+=======
+>>>>>>> parent of d0002ef6... Implementation of -lp alias for -LiteralPath variable #6732
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Rename-Item.Tests.ps1]: Please fix.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Set-Content.Tests.ps1 [hunk 6]
```diff
@@ -15,6 +15,13 @@ Describe "Set-Content cmdlet tests" -Tags "CI" {
         $contentWarning.Message | Should -Match "-AsByteStream"
     }
 
+<<<<<<< HEAD
+    It "should not throw exception when -LP alias for LiteralPath is used " {
+        { "test" | Set-Content -lp "${TESTDRIVE}\testfile.txt" -WhatIf } | Should -Not -Throw
+    }     
+
+=======
+>>>>>>> parent of d0002ef6... Implementation of -lp alias for -LiteralPath variable #6732
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Set-Content.Tests.ps1]: Please fix.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Add-Type.Tests.ps1 [hunk 7]
```diff
@@ -153,7 +153,15 @@ public class AttributeTest$guid : PSCmdlet
...
         $returnedTypes2 = Add-Type -LiteralPath $file1,$file2 -PassThru
+<<<<<<< HEAD
+
+        $returnedTypes[0].Assembly.FullName | Should BeExactly $returnedTypes2[0].Assembly.FullName
+
+        # Also check that -lp alias of '-LiteralPath' works.
+        { $returnedTypes3 = Add-Type -lp $file2 -PassThru } | Should -Not -Throw        
+=======
         $returnedTypes[0].Assembly.FullName | Should BeExactly $returnedTypes2[0].Assembly.FullName
+>>>>>>> parent of d0002ef6... Implementation of -lp alias for -LiteralPath variable #6732
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Add-Type.Tests.ps1]: Please fix.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Tee-Object.Tests.ps1 [hunk 8]
```diff
@@ -6,6 +6,25 @@ Describe "Tee-Object" -Tags "CI" {
...
+            Remove-Item $teefile -ErrorAction SilentlyContinue
+        }
+
+        It "Should tee the output to a file" {
+            $teefile = $testfile
+            echo teeobjecttest3  | Tee-Object $teefile
+            Get-Content $teefile | Should -BeExactly "teeobjecttest3"
+            Remove-Item $teefile -ErrorAction SilentlyContinue
+        }
+=======
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Tee-Object.Tests.ps1]: Please fix.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Tee-Object.Tests.ps1 [hunk 9]
```diff
@@ -19,6 +38,7 @@ Describe "Tee-Object" -Tags "CI" {
 	        Get-Content $teefile | Should -BeExactly "teeobjecttest3"
 	        Remove-Item $teefile -ErrorAction SilentlyContinue
 	    }
+>>>>>>> parent of d0002ef6... Implementation of -lp alias for -LiteralPath variable #6732
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Tee-Object.Tests.ps1]: Please fix.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Add-Type.Tests.ps1 [hunk 10]
```diff
@@ -153,7 +153,7 @@ public class AttributeTest$guid : PSCmdlet
         # Return the same assembly if source code has not been changed.
         # Also check that '-LiteralPath' works.
         $returnedTypes2 = Add-Type -LiteralPath $file1,$file2 -PassThru
-        $returnedTypes[0].Assembly.FullName | Should BeExactly $returnedTypes2[0].Assembly.FullName
+        $returnedTypes[0].Assembly.FullName | Should BeExactly $returnedTypes2[0].Assembly.FullName
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Add-Type.Tests.ps1]: Please remove trail spaces.

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Tee-Object.Tests.ps1 [hunk 11]
```diff
@@ -6,6 +6,20 @@ Describe "Tee-Object" -Tags "CI" {
...
 	$testfile = Join-Path -Path (Join-Path -Path $PSScriptRoot -ChildPath assets) -ChildPath testfile.txt
 
+        It "Should return the output to the screen and to the variable" {
+            $teefile = $testfile
+            echo teeobjecttest1 | Tee-Object -variable teeresults
+            $teeresults         | Should -BeExactly "teeobjecttest1"
+            Remove-Item $teefile -ErrorAction SilentlyContinue
+        }
+
+        It "Should tee the output to a file" {
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Tee-Object.Tests.ps1]: Seems this two test don't related to the PR.

Comment (kvprasoon) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/Tee-Object.Tests.ps1]: @iSazonov yes and those are existing .

Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/Tee-Object.Tests.ps1]: You should keep the commit history and do rebase more accurately.

Comment (kvprasoon) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/Tee-Object.Tests.ps1]: (y) Sure and I'm learning :-)

Comment (kvprasoon) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/Tee-Object.Tests.ps1]: Yes, as of now all changes are committed.

**Comment by TravisEz13** [comment]
I think I fixed the one test you have in my master branch here:  https://github.com/travisez13/PowerShell ,but why did you remove the rest of the tests?

**Comment by kvprasoon** [comment]
@TravisEz13 All the tests ?, If you are referring tests for the new alias, then I've initially added tests for `-lp` alias for all the cmdlets, iSazonov asked me to remove all the tests I added for `-lp` alias, hence removed.

please correct me if I'm wrong.

**Comment by TravisEz13** [comment]
ok... I can clean up the PR better with that knowledge... give me a bit.

**Comment by TravisEz13** [comment]
I squashed the changes down to one commit with the tests removed.

**Comment by kvprasoon** [comment]
@TravisEz13 Thanks and how did you do that ? I don't know how to do it.

**Comment by TravisEz13** [comment]
@kvprasoon Very carefully...
I created a commit (this took two tries) that reverted the change to test that wasn't completely reverted and just squashed the changes.  I then changed the author back to you (you wouldn't have to do that.)

**Comment by TravisEz13** [comment]
@kvprasoon  Can you ping me when the tests finish?

**Comment by kvprasoon** [comment]
@TravisEz13 of course

**Review by iSazonov — APPROVED** [review]

**Comment by kvprasoon** [comment]
@TravisEz13 Completed with one failure(failed for a test case for Test-Connection) in appveyor CI

**Comment by TravisEz13** [comment]
restarted appveyor to get clean results

**Comment by SteveL-MSFT** [comment]
@TravisEz13 committee review is just to accept the new alias?

**Comment by daxian-dbw** [comment]
@SteveL-MSFT It's a change that affects many cmdlets, so I think committee might want to take a quick look.

**Comment by daxian-dbw** [comment]
@PowerShell/powershell-committee has reviewed and agreed to add the alias `-LP` for the parameter `-LiteralPath`.

**Pull request merged**